res/controllers/Hercules_DJControl_Jogvision-scripts.js

```diff
@@ -0,0 +1,310 @@
+// - Enabled Jogwheel Outer LED rotation
+// - Enabled Beat LEDs
+// 
+// Changes to v1.1
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+// 
+// v1.0 : Original release
+//
+// ****************************************************************************
+function DJCJV() {}
+var DJCJV = {};
+
+var on = 0x7F;
+var off = 0x00;
+var alpha = 1.0 / 8;
+var beta = alpha / 16;
+
+DJCJV.scratchButtonState = true;
+DJCJV.beatAccumDA = 1;
+DJCJV.beatAccumDB = 1;
+DJCJV.rotationA = 0x00;
+DJCJV.rotationB = 0x00;
+DJCJV.ledRotationTimer = 0;
+
+function dec2hex(dec) {
+    return '0x' + (dec + 0x10000).toString(16).substr(-4).toUpperCase();
+}
+
+// Initialization
+DJCJV.init = function() {
+
+    //Set all LED states to off
+    midi.sendShortMsg(0xB0, 0x7F, off);
+    midi.sendShortMsg(0xB1, 0x7F, off);
+
+    // Set Vinyl button LED On.
+    midi.sendShortMsg(0x90, 0x45, on);
+    DJCJV.scratchButtonState = true;
+    midi.sendShortMsg(0x90, 0x46, on);
+
+    // Set Headphone CUE/MIX LED state
+    if (engine.getValue("[Master]", "headMix") > 0.5) {
+        midi.sendShortMsg(0x90, 0x4C, on);
+    } // headset "Mix" button LED
+    else {
+        midi.sendShortMsg(0x90, 0x4D, on);
+    } // headset "Cue" button LED
+
+    //Enable Soft takeover
+    engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[QuickEffectRack1_[Channel1]]", "super1", true);
+    engine.softTakeover("[QuickEffectRack1_[Channel2]]", "super1", true);
+
+    //Set effects Levels - Dry/Wet - Filters
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect1]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect2]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect3]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect2]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect3]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1]", "mix", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2]", "mix", 0.6);
+    engine.setParameter("[QuickEffectRack1_[Channel1]]", "super1", 0.5);
+    engine.setParameter("[QuickEffectRack1_[Channel2]]", "super1", 0.5);
+
+    // Connect the VUMeters and Jog Inner LED
+    engine.connectControl("[Channel1]", "VuMeter", "DJCJV.vuMeterUpdateDA");
+    engine.connectControl("[Channel2]", "VuMeter", "DJCJV.vuMeterUpdateDB");
+    engine.connectControl("[Channel1]", "playposition", "DJCJV.wheelInnerUpdateDA");
+    engine.connectControl("[Channel2]", "playposition", "DJCJV.wheelInnerUpdateDB");
+
+    // Enable wheels outer leds rotation by timer (when channel is playing)s
+    DJCJV.ledRotationTimer = engine.beginTimer(20, function() {
+        if (engine.getValue("[Channel1]", "play") == 1) {
+            midi.sendShortMsg(0xB0, 0x60, dec2hex(DJCJV.rotationA));

```

Comment:
Doesn't `dec2hex`  return a string instead of a single byte?

Reply:
Yeah, there is no need to convert between decimal and hexidecimal. A number is a number. You can write `0` or `0x00`; they're treated the same.